native/spark-expr/src/cast.rs

```diff
@@ -1746,35 +1844,92 @@ mod tests {
     #[test]
     #[cfg_attr(miri, ignore)] // test takes too long with miri
     fn timestamp_parser_test() {
+        let tz = &timezone::Tz::from_str("UTC").unwrap();
         // write for all formats
         assert_eq!(
-            timestamp_parser("2020", EvalMode::Legacy).unwrap(),
+            timestamp_parser("2020", EvalMode::Legacy, tz).unwrap(),
```

Comment:
Could you also add tests for 3-digit year

Reply:
I've covered the test-cases for 3 digit years from L1877 - 1904, is there anything that I'm missing out?